xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/FormContainerElement.java

```diff
@@ -73,20 +73,24 @@ protected WebElement getFormElement()
 
     public void fillFieldsByName(Map<String, String> valuesByNames)
     {
-        Map<WebElement, String> valuesByElements = new LinkedHashMap<>((int) (valuesByNames.size() / 0.75));
-
+        Map<WebElement, String> valuesByElements = new LinkedHashMap<>();
+        
+        String lastElementName = null;
         for (String name : valuesByNames.keySet()) {
             valuesByElements.put(getFormElement().findElement(By.name(name)), valuesByNames.get(name));
+            lastElementName = name;
         }
         fillFieldsByElements(valuesByElements);
         
         /* Register password confirmation is usually the last element that needs to be validated by liveValidation.
           This wait allows to solve a race condition between the form submission and the computation of the status of
           those fields. We force the status to be solved before we try anything else, especially submitting the form.
+          Unfortunately in Java17 we do not have lastEntry(), so we use a few non optimized operations instead. 
+          This is okay because the Map should not contain a lot of elements.
           */
-        if (valuesByNames.containsKey("register2_password")) {
-            getDriver().waitUntilCondition(driver -> !getFormElement().findElement(By.name("register2_password"))
-                .getAttribute(CLASS_ATTRIBUTE).isEmpty());
+        if(valuesByElements.size() > 0) {
```

Comment:
You should probably use `!valuesByElements.isEmpty()` here, I'm surprised you don't have a suggestion to change it in the IDE.

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/3675/commits/a45c50db7fa666e16a90da1b1606940b44bd4971 :+1: 
I did get a suggestion but I overlooked it because of other suggestions of `maybe NullPointerException` just after.